src/server/replica.cc

```diff
@@ -226,6 +226,23 @@ void Replica::Pause(bool pause) {
   sock_->proactor()->Await([&] { is_paused_ = pause; });
 }
 
+std::error_code Replica::TakeOver(std::string_view timeout) {
+  VLOG(1) << "Taking over";
+
+  std::unique_lock lk(sock_mu_);
```

Comment:
currently we use the mutex only in the connect and close of socket. Why do we need it here? worse adding a comment

Reply:
good point, probably don't need it.